sdk/core/azure-core-amqp/src/amqp/management.cpp

```diff
@@ -131,9 +131,9 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
       if (rv != _internal::ManagementOpenStatus::Ok)
       {
         Log::Stream(Logger::Level::Warning) << "Management operation failed to open.";
-        m_messageSender->Close();
+        m_messageSender->Close(context);
         m_messageSenderOpen = false;
-        m_messageReceiver->Close();
+        m_messageReceiver->Close(context);
         m_messageReceiverOpen = false;
```

Comment:
Is this exception- and thread-safe?
I think I at least saw some throw expressions in these Close methods' implementations. Maybe it is ok.